### ISSUE TITLE
[RFC] Project restructuring

### ISSUE THREAD

**Issue by bb010g** [TheOriginalSINe/Project-M-CC#398]
# Proposal

Project M Community Complete's core goal is to bring the leaked newcomers, completed in a style as close as possible to Project M Development Team's 3.6 work, to players in an accessible, unintrusive manner.

Project M Community Complete's secondary goal is to complete content that the Project M Development Team strongly intended to complete at the same level of quality and bring that content to players in an accessible, unintrusive manner.

The project will focus mainly on completing Project M + BrawlEx (PMEX) versions of the newcomers. To support this, an accessible way to get PMEX with PMCC characters will be provided. Ideally, tooling would be made to support trivial addition to any PMEX build.

To maintain accessibility, content created by CC will have all the essential resources for a piece of content stored in a respective directory in a manner allowing for easy changes to the source and the quick creation of PM-friendly files from that (modified) source.

To clearly separate the more controversial newcomer-related portions of PMCC and the more respected 3.61 or earlier–related portions of PMCC, custom content will be separated into at least two repositories on GitHub.

To support the creation of "official" PMCC builds and provide what the community currently lacks, PMCC will provide an accessible way to get a stable 3.61 Project M + BrawlEx build with no obvious PMCC branding.
# Explanation

I'm sure everyone has heard about [this post](https://web.archive.org/web/20160417054820/https://www.reddit.com/r/SSBPMCE/comments/4d2fzz/cc_broke_my_wii/) by now, discussing how their Wii has problems loading Brawl mods now thanks to what appears to be a fault in the current CC codeset. If we stick with the current code base, we're going to have an insanely hard time escaping that stigma, even if we manage to fix the bug(s). People just won't try out CC. This is the nice project-could-die argument for this restructure.

Some people in the Brawl modding community don't like what we're doing, which is understandable. We're in an odd spot with the subject we're dealing with, our personal dynamic, and the speed we progress at. Allowing for the use of CC by users themselves in any build they want easily would be a great gain for the player, by not feeling restricted if they want to use CC content, for the build creator, by not having to make decisions on whether they want to associate with CC, and for us, by getting more hands on the content and more feedback.

We spread the original scope too broadly. Everyone loves a goal that promises them the world they wished for, but that's hard to get, and as I said, we have problems. Having a concrete core goal will provide newcomers with a direction on what they should be working on and let them know whether they would be a good fit at that moment. The other secondary content—team alts, new stage selection, retro mode—is just that, _secondary_. It's nice to have. The PMDT was planning it, but it's nowhere near as important for this project and valuable to the community as the newcomers.

PMEX is stable, well-made, and only going to get more popular as time goes on with the finality of PM. We should take full advantage of it and avoid NIH syndrome. Who cares if they're on the clone engine or not? Do they work well? Can I put them where I want? Technical feats are cool for technical people. PMEX is more useful than the clone engine short and long term.

This section may be expanded as time goes on and the RFC changes to better reflect the proposal or explain murky areas better. Proposal updates will be noted in their own section here if made.

---

If you have any questions about this, please ask. The sooner we figure this out, the sooner we can stop spinning our wheels as much as a project. If you support this, use the :+1: reaction. If you don't, use the :-1: reaction. Don't leave a comment if you have nothing to add to this, it just clogs up the issue. If you disagree, you probably have something to add, so please comment and/or make an appropriate reaction to a comment that reflects your sentiments.

**Comment by icecapman10** [comment]
So, your using Brawl EX for the newcomers?

**Comment by bb010g** [comment]
@icecapman10 Yes, as stated in paragraph 3.

**Comment by bb010g** [comment]
Approved by @Rage17.

**Comment by bb010g** [comment]
Approved by @LnktheWolf.

**Comment by icecapman10** [comment]
Ok, I'm fine with it

**Comment by MlKe12** [comment]
It's gonna work on Dolphin ?

**Comment by cimpyhigh** [comment]
This is a great idea! It was all I was planning on using from the build anyway (with the exception of retro mode).

**Comment by bb010g** [comment]
@Mariokirby123 Yes.

**Comment by bb010g** [comment]
cc/ @bringobrongo @OvisCantus @RikiDVX @thecoolguy27 @vinegar97

**Comment by ithunderHD** [comment]
im in for the this idea bb010g,the thing is there are no  people working on the project lately and,i will be happy to do all I can to contribute. Its just been busy for the last few months on this project and I haven't been able to do much. but now im free so we really need to keep this project going so we don't let down anybody who gave fate to this project but for now  I agree on this situation and we must focus on this problem. I been thinking about this project  for along time ever sense December,1st 2015 and we cant quit now like pm team did , even the new pm team quit. but we cant afford to quit because we have come so far eversense  that day pm was discontinued it. pm has meaned a lot to me in different ways and others too. so In order to get this project working we must work as team to get this done. Ive seen confusion on gitterchannel like arguments. Basically this project is on the verge of being ceased so we must work  as a team even with low number of people we have now. this is a speech I made constanting of my thoughts about the project I hope this was  inspirational to everyone.

**Comment by robosteven** [comment]
Even if there are bugs, stability is extremely important to this project. I'm 100% in favor of the change.

**Comment by bb010g** [comment]
All responses I've seen so far have been positive. I propose approval in one week (2016-04-27), if no one disapproves.

**Comment by bb010g** [comment]
No disapproval has been stated here, so this RFC is now approved. This issue will be closed when necessary actions have been taken in the repositories and issues to complete the transition.

@Yohan1044 has suggested waiting for a cleaned up PMEX build before switching over, but preparations and testing can be started now.

**Comment by Yohan1044** [comment]
As far as I can recall, I've never advocated for PMCC moving to PM+EX. I've only mentioned pros and cons of the change, and factors to consider. I think I actually once advocated for sticking to the original GCT, as it aligns more with the initial scope and goals of the project and isn't as disapproved by the God of Codes.
End of the day, I am not passionate enough about the outcome of this RFC to argue/vote for any particular side.

Edit: I do distinctly recall suggesting asking for Pyotr's permission. He's against the change, but acknowledges that his contributions of the PM community are open to all, and he will not specifically forbid PMCC from using his works.

**Comment by Creepyshadow64** [comment]
Am new here and when will it be released

**Comment by GunBlazer** [comment]
Well, in all honesty, I don't think it would be a good idea, but since it has already been approved...

Also, I've never experienced that corruption linked at the OP, ever since I got the first version of this project.

**Comment by ghost** [comment]
I've stumbled across the reddit post mentioned in the Explanation section before, and when I initially saw it my first reaction was that it was some random noob having a very isolated problem (reading GBAtemp for a while you get used to ignoring posts like that). Not sure how far the bricking stigma goes, but I've never had any trouble with Project M CC and can't wait to see where it goes.

**Comment by Delta-Kuro** [comment]
As long as their move sets won't be majorly changed, I'm absolutely fine with it.

**Comment by TheSmashBoy** [comment]
👍🏼👍🏼👍🏼👍🏼👍🏼 I approve of this restructuring. As @Delta-Kuro stated, as long as it doesn't affect the character movesets it's fine.

 P.S: Recently signed. Just thought I laid my opinion on this.

**Comment by TheSmashBoy** [comment]
Hey @bb010g are you guys currently working on the project, character animations and movesets? You guys haven't uploaded nor updated the current build from March 6 since then. What's going on?

**Comment by omegamalkior** [comment]
He replied to post on reddit I made on reddit (SSBPMCE I think it's called) regarding the status of PMCC.
Here's a link if you want to see:
https://www.reddit.com/r/SSBPMCE/comments/4tybck/is_pmcc_stilled_being_worked_on_and_if_so_where/

**Comment by TheSmashBoy** [comment]
Thanks @omegamalkior , I'll see to it now.

**Comment by TheSmashBoy** [comment]
@omegamalkior I read the link. It's just as I have suspected. It's not only the restructuring initiative but the polishing of the characters. This was very helpful. A lot of people are saying you guys are dead without any substance. This link has helped by a lot. 

 P.S: I recently found a PSA from U.N Owen for Isaac and some Sami recolors from Ran_ (a YouTuber). It's on #421. I explain everything there. Thought some of you could check it out and write some feedbacks.

**Comment by MrGameCrazy** [comment]
I noticed that there was an update recently for PM CC. What has changed since the last update?

**Comment by TheSmashBoy** [comment]
@MrGameCrazy Well for starters. I pull requested U.N Owen Isaac and Sami recolors, plus a few menu fixes. I'm currently waiting for bb010g to re-upload it.

 Also opened up new repos for each character. They are in their PMEX format with no soundbank sadly. He is currently working on how to fix that.

**Comment by MrGameCrazy** [comment]
Does the current PM CC build use PMEX right now?

**Comment by TheSmashBoy** [comment]
@MrGameCrazy No. But bb010g is working on that. Only the characters are in PMEX format. Which means you need to download your own PMEX custom-free and test them. Maybe even start working on their animations. Until the upstream repo gets updated to PMEX format.

**Comment by MegAmi24** [comment]
... So what does this all mean?
